modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/typeclasses/MonadDefer.kt

```diff
@@ -44,9 +44,13 @@ interface MonadDefer<F> : MonadThrow<F>, Bracket<F, Throwable> {
 
   fun lazy(): Kind<F, Unit> = delay { }
 
-  fun <A> deferUnsafe(f: () -> Either<Throwable, A>): Kind<F, A> =
+  fun <A> delayUnsafe(f: () -> Either<Throwable, A>): Kind<F, A> =
```

Comment:
What is unsafe about this operation? If it were unsafe in any case it should be placed under the `arrow.unsafe` receiver to be consistent with other unsafe api's and runners post 0.9.0

Reply:
It's called unsafe because the function inside can fail with a Throwable, and that's forwarded via the left side of the Either. The way I see it, it's a "softer" version of flatMap for non-effectful code that can still fail.

Reply:
When MonadDefer gets parametrized to `E`, this'd be `f: () -> Either<E, A>`

Reply:
but if it takes already a Throwable inside the either and defer itself is safe this isn´t unsafe.
I think a better name for this is needed. This is also what `effect` takes in `fx` for what will become `BIO`. I think this is just a lazy version of `fromEither` and I'd call it `deferOrRaise`. It's not unsafe because it does not perform effects when invoked.

Reply:
I'm okay with `delayOrRaise`, yep. @juliankotrba, opinions?

Reply:
yeah, delayOrRaise is a better name

Reply:
delayOrRaise sounds good, should this be changed in a new PR?